src/select/styled-components.js

```diff
@@ -33,6 +33,9 @@ export const Input = styled('input', props => {
   return {
     ...getInputStyles({...props, $size: SIZE.default}),
     cursor: 'pointer',
+    ':hover': {
+      cursor: 'pointer',
+    },
```

Comment:
Is there any reason to nest `cursor` rules inside of `:hover` anywhere in our codebase? It looks like here you're doing it to override a style from `Input`, but maybe we should just update the Input component styles to not use `:hover`.

Reply:
another way is to override `$disabled` prop here, what I did right now

Reply:
Just curious–why is that a better choice? Wouldn't you still want some of the other disabled input styles (like text color and placeholder color)?

Is moving `cursor` rule out of `:hover` inside of `getInputStyles` not an option?

Reply:
No, it's not. Now those two styles are compromised in Select. Changing code in original Input implementation just to comply with Select is a violation of inheritance. Input shouldn't know about needs of Select as long as Select can cope with it.

Reply:
I don't think that is necessarily true. We developed most of these components in a silo and if the change makes reasonable sense when it comes to integrating with other components then it's worth making. Otherwise, our users will end up having to jump through the hoops just like we did now. That's speaking generally. But it is weird to see a `:hover` rule with `:cursor` inside of it. Maybe we can open a separate task for that?

Reply:
+1 abmai. 

We should be consistent about how we define `cursor` rules throughout baseui, and it seems pretty widely accepted to not nest them in `:hover`. In reality we're fixing Input, and the side effect of this is that Select (and probably many other components) also benefit.

Reply:
ok, this code is no longer actual